Feature/160 create test data

kevinwallimann opened the pull request:
**How to use**
1. Create a local database on port 5432 named "hyperdriver" and a user also named "hyperdriver" (See `LocalDatabaseConfig` for config)
2. Execute initial sql-script `db_script_v1.sql`
3. Execute test `TestDataGeneratorLocal` (in IntelliJ). If given the choice, execute with scalatest, not Junit.
4. Play around
5. Truncate all tables like this `truncate event, job_instance, job_definition, dag_instance, dag_definition, sensor, workflow;`
6. Repeat from step 3

Let me know if the handling is acceptable, or you wish to execute from cmdline, nicer random strings etc.

kevinwallimann committed:
wip

kevinwallimann committed:
Testdata generator for workflows

kevinwallimann committed:
Add test case to insert run data

kevinwallimann committed:
Remove comment

jozefbakus commented on src/test/scala/za/co/absa/hyperdrive/trigger/testdata/LocalDatabaseConfig.scala:
```diff
@@ -0,0 +1,29 @@
+trait LocalDatabaseConfig extends BeforeAndAfterAll { this: Suite =>
+
+  override def beforeAll(): Unit = {
+    System.setProperty("db.driver", "org.postgresql.Driver")
```
Is it possible to use application properties file instead of this ? (I mean, without this class).

kevinwallimann replied:
Not entirely. From the spring side, I can use `@TestPropertySource` to define another properties file to use than the default one. However, slick obviously doesn't know about this. By default, Slick loads any configuration file with name `application.conf`, `application.json` or `application.properties` which is on the classpath. (See `JdbcBackend.forConfig`, `DefaultConfigLoadingStrategy.parseApplicationConfig`, `SimpleIncluder.fromBasename`). 

An alternative properties file can be specified with the system properties `config.resource`, `config.file` or `config.url` (which are unfortunately very general names, could easily be conflicted imo). So I could extract the three properties to a separate file, but I would still have to define the above mentioned system property.

jozefbakus commented on src/test/scala/za/co/absa/hyperdrive/trigger/TestUtils.scala:
```diff
@@ -0,0 +1,28 @@
+import scala.concurrent.duration.Duration
+
+object TestUtils {
+  def await[T](future: Future[T]): T = {
```
It would be fine to replace all occurrences of this in tests. We can do it later.

kevinwallimann replied:
Yes I didn't want to touch all those tests in this PR

jozefbakus replied:
Ok

jozefbakus reviewed (commented):
Random strings are fine. But I execute it via cmdline would be perfect.
+ Would be fine to have different sensors and job types data.

kevinwallimann committed:
Add maven profile for testdata generation, testdata for shell script and time based sensor

kevinwallimann commented:
> Random strings are fine. But I execute it via cmdline would be perfect.
> 
> * Would be fine to have different sensors and job types data.

Updated PR
You can do now: `mvn test -Ptestdata` to create the testdata. Also added different sensor and job type

kevinwallimann committed:
Merge branch 'feature/new-ui' into feature/160-create-test-data

kevinwallimann committed:
Fix merge errors

kevinwallimann committed:
Fix wrong parameters for jobs

jozefbakus reviewed (approved):
👍🏻
